File: source/quantum-script-libstdobject.cpp
//
// Quantum Script
//
// Copyright (c) 2020 Grigore Stefan <g_stefan@yahoo.com>
// Created by Grigore Stefan <g_stefan@yahoo.com>
//
// MIT License (MIT) <http://opensource.org/licenses/MIT>
//

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "quantum-script-libstdfunction.hpp"

#include "quantum-script-libstdobject.src"

//#define QUANTUM_SCRIPT_DEBUG_RUNTIME

namespace Quantum {
	namespace Script {

		namespace LibStdObject {

			void initExecutive(Executive *executive, void *extensionId) {
				executive->compileStringX(libStdObjectSource);
			};

		};
	};
};

File: source/quantum-script.lib/quantum-script-libstdobject.js
//
// Quantum Script
//
// Copyright (c) 2020-2022 Grigore Stefan <g_stefan@yahoo.com>
// Created by Grigore Stefan <g_stefan@yahoo.com>
//
// MIT License (MIT) <http://opensource.org/licenses/MIT>
//

Object.prototype.create = function(prototype_) {
	var tmp_ = function() {};
	tmp_.prototype = prototype_;
	return new tmp_();
};
